### PR TITLE
Fix minor errors in analyzer.py

### PR DESCRIPTION
**Pull request by havanagrawal**
Some errors that would not work if the respective code was run.

Ignore the whitespace differences.

Important changes on L320, L791, L810, L988

### CODE REVIEW

File: spotpy/analyser.py [hunk 1]
```diff
@@ -307,26 +307,26 @@ def get_best_parameterset(results,maximize=True):
...
-    
+    :type: class
+
     :return: Possible minimal and maximal values of all parameters in the parameters function of the spotpy_setup class
     :rtype: Two arrays
     """
-    randompar=spotpy_setup.parameters()['random']        
+    randompar = spotpy_setup.parameters()['random']
     for i in range(1000):
-        randompar=np.column_stack((randompar,self.parameter()['random']))
```
Comment (havanagrawal) [inline spotpy/analyser.py]: self is undefined here

File: spotpy/analyser.py [hunk 2]
```diff
@@ -964,34 +964,34 @@ def plot_bestmodelruns(results,evaluation,algorithms=None,dates=None,ylabel='Bes
...
-    for i in range(len(results)):       
+    for i in range(len(results)):
         if calculatelike:
             likes=[]
             sim=get_modelruns(results[i])
             par=get_parameters(results[i])
             for s in sim:
-                likes.append(spotpyobjectivefunctions.lognashsutcliff(evaluation,list(s)))                                
-                                
+                likes.append(spotpy.objectivefunctions.lognashsutcliffe(evaluation,list(s)))
```
Comment (havanagrawal) [inline spotpy/analyser.py]: Incorrect spelling of `lognashsutcliffe`

File: spotpy/analyser.py [hunk 3]
```diff
@@ -964,34 +964,34 @@ def plot_bestmodelruns(results,evaluation,algorithms=None,dates=None,ylabel='Bes
...
-            
+
         else:
             if objectivefunctionmax==True:
                 index,maximum=get_maxlikeindex(results[i])
             else:
                 index,maximum=get_minlikeindex(results[i])
             bestmodelrun=list(get_modelruns(results[i])[index][0])#Transform values into list to ensure plotting
-        
-        maxLike=spotpy.objectivefunctions.lognashsutcliff(evaluation,bestmodelrun)
```
Comment (havanagrawal) [inline spotpy/analyser.py]: Incorrect spelling of `lognashsutcliffe`

File: spotpy/analyser.py [hunk 4]
```diff
@@ -807,7 +807,7 @@ def plot_posterior(results,evaluation,dates=None,ylabel='Posterior model simulat
 
     parameternames=list(get_parameternames(results)    )
     bestparameterstring=''
-    maxNSE=spotpy.objectivefunctions.nashsutcliff(bestmodelrun,evaluation)
```
Comment (havanagrawal) [inline spotpy/analyser.py]: Incorrect spelling of `nashsutcliffe`

File: spotpy/analyser.py [hunk 5]
```diff
@@ -788,14 +788,14 @@ def plot_posterior(results,evaluation,dates=None,ylabel='Posterior model simulat
     if sort:
         results=sort_like(results)
     if calculatelike:
-        likes=calc_like(results)
```
Comment (havanagrawal) [inline spotpy/analyser.py]: Missing argument for evaluation

**Comment by thouska** [comment]
Good job, thank you very much!

**Pull request merged**